### PR TITLE
Implemented flake8 and desired settings in setup.cfg

### PR DESCRIPTION
**Pull request by Daniel-Jang**
To check the entire codebase for pep 8 violations, simply type:

flake8 

in the console. If nothing shows up, that means there are no violations.

Relevant issue: https://github.com/cass-degrees/CASS-Degrees-Code/issues/40

### CODE REVIEW

File: cassdegrees/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[flake8]
+# F401: Ignore unused imports - some files are automatically generated with import statements.
+# F403, F405: Ignore warnings caused by * imports
+ignore = F401, F403, F405
+# Ignore the git files and migration files that are out of our control.
+exclude = .git, *migrations*
+# Set max length to be the same as what PyCharmm currently has it at (120).
+max-line-length = 120
```
Comment (j-selby) [inline cassdegrees/setup.cfg]: Newline.

**Comment by j-selby** [comment]
Hmmm. Would this be better serviced with a GitHub app (e.g. https://github.com/marketplace/actions/python-lint ) so we don't have to manually do this every time?

**Comment by j-selby** [comment]
Specifically https://github.com/marketplace/actions/python-syntax-checker might be what we are looking for here.

**Comment by Daniel-Jang** [comment]
This allows us to check for code violations and fix all of them before commit. 

If we go your method only, we'd have to commit our code, look at status of CI, come back to edit all the changes, commit again, check if further issues popped up and repeat until no pep 8 errors remain.

What I suggest is that we have both this and the CI. How does that sound?

**Comment by j-selby** [comment]
Nope, that sounds perfectly fine! Just tested it out as above, but it will need a bit of configuration

**Comment by Daniel-Jang** [comment]
Did you mean the local flake8? If yes, here it is: https://github.com/cass-degrees/CASS-Degrees-Code/pull/54/commits/42d1d165a2dd34dcd422f00cafe6be78f5e0c628#diff-9a1f66670f0a343dc88bff1ada2c310d
I added comments on the reasoning as well.

If you meant the CI pep 8 checker, I will try and see how I can add configurations to it. I'll be adding the exact configurations as I had it with the local flake8.
(note to self: https://pep8speaks.com/)

**Comment by j-selby** [comment]
Yeah. I meant the CI - should just be an additional configuration file to merge here.

**Review by JackSekoranja — APPROVED** [review]
Ran it and it works, although note that it should be run in the `CASS-Degrees-Code/cassdegrees` directory

**Comment by Daniel-Jang** [comment]
Ok, I added the .pep8speaks.yml configuration file with some settings.

File: .pep8speaks.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+scanner:
+  diff_only: True
+  linter: flake8
+
+flake8:
+  max-line-length: 120  # Set max length to be the same as what PyCharmm currently has it at (120).
+  ignore:
+    - F401  # Ignore unused imports - some files are automatically generated with import statements.
+    - F403  # F403, F405: Ignore warnings caused by * imports
+    - F405
```
Comment (j-selby) [inline .pep8speaks.yml:10]: Newline :P

**Review by j-selby — APPROVED** [review]
Looks good!

File: .pep8speaks.yml (hunk 2 above)
Comment (Daniel-Jang) [reply .pep8speaks.yml:10]: Oh man, I noticed this comment after merging it...

Haha damn sorry again.... I'll fix it in the next branch I work on :/

**Pull request merged**